ogr/ogrsf_frmts/adbc/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+add_gdal_driver(
+  TARGET ogr_ADBC
+  SOURCES
+      ogr_adbc.h
+      ogradbcdriver.cpp
+      ogradbcdataset.cpp
+      ogradbclayer.cpp
+  CORE_SOURCES ogradbcdrivercore.cpp
+  PLUGIN_CAPABLE
+  NO_SHARED_SYMBOL_WITH_CORE
+  STRONG_CXX_WFLAGS)
+
+if(NOT TARGET ogr_ADBC)
+    return()
+endif()
+
+gdal_standard_includes(ogr_ADBC)
+target_include_directories(ogr_ADBC PRIVATE $<TARGET_PROPERTY:ogrsf_generic,SOURCE_DIR>)
+gdal_target_link_libraries(ogr_ADBC PRIVATE AdbcDriverManager::adbc_driver_manager_shared)
```

Comment:
I'm a bit hesitant on that:
- that would mean extra code to maintain in GDAL
- When we vendor things, we make an effort to rename their symbols, which is an extra layer of maintainance on GDAL side. If adbc_driver_manager code itself has some provision for a #define ADBC_DRIVER_MANAGER_NAMESPACE that could allow to prefix its symbols with a user specified prefix, that would save that effort on GDAL side.
- what about the Windows side of things? Initially, I tried to use Conda to test on Windows on our CI, but it failed, and then I realized that
https://github.com/conda-forge/arrow-adbc-split-feedstock/blob/6f2af5a9b0de9cbdf895ec0ab1e516c306abac67/recipe/meta.yaml#L31 disabled Windows packaging. Is it a Conda thing, or is there's something more fundamental in adbc_driver_manager itself that prevents it from working on Windows?
- if we vendor, what about the stability of the interface between adbc_driver_manager and ADBC drivers themselves? I mean if we forget to update the vendored  adbc_driver_manager, will that affect its working with ADBC drivers that would have been built against a more recent external ADBC interface?
- with respect to my above symbol renaming, do ADBC drivers link with adbc_driver_manager itself? If so, then symbol renaming is not possible, and vendoring would be a bit risky.

Reply:
> there won't be widely available linux packages anytime soon.

IMHO, that's something to consider in the top of the priority list for success of ADBC rather than investing in all other workarounds.

> If it worked using `FetchContent` to an official ADBC release would that be sufficient?

not really, at least for GDAL as packaged by Linux distributions. I believe some build systems disable network after they have downloaded the tarball and installed all dependencies to make sure it is fully reproducible